Fix hashing context example

dmipeck opened the pull request:
The example function produces invalid hashes for files that aren't a multiple of `CHUNK_SIZE` bytes in size

Fixes two issues:

+ replaces use of `eof_reached()` who's docs say to use `file.get_position() < file.get_len()` instead when looping
+ uses remaining number of bytes instead of `CHUNK_SIZE` if `CHUNK_SIZE` would be too large

<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

raulsntos commented on doc/classes/HashingContext.xml:
```diff
@@ -11,46 +11,52 @@
+			# Open the file to hash.
+			var file = FileAccess.open(path, FileAccess.READ)
+			# Update the context after reading each chunk.
+			while not file.get_position() &lt; file.get_len():
+				var remaining = file.get_len() - file.get_position()
```
The method is called `get_length`.

```suggestion
			while not file.get_position() &lt; file.get_length():
				var remaining = file.get_length() - file.get_position()
```

raulsntos commented on doc/classes/HashingContext.xml:
```diff
@@ -11,46 +11,52 @@
+			// Update the context after reading each chunk.
+			while (file.GetPosition() &lt; file.GetLength())
+			{
+				int remaining = file.GetLen() - file.GetPosition();
```
These methods return `ulong` so you need to cast the result to `int`.

```suggestion
				int remaining = (int)(file.GetLength() - file.GetPosition());
```

raulsntos commented on doc/classes/HashingContext.xml:
```diff
@@ -11,46 +11,52 @@
+			while (file.GetPosition() &lt; file.GetLength())
+			{
+				int remaining = file.GetLen() - file.GetPosition();
+				if (remaining &gt; ChunkSize) {
```
C# code-style is to add a new line before the curly bracket.

```suggestion
				if (remaining &gt; ChunkSize)
				{
```

raulsntos commented on doc/classes/HashingContext.xml:
```diff
@@ -11,46 +11,52 @@
+				if (remaining &gt; ChunkSize) {
+					remaining = ChunkSize;
+				}
+				ctx.Update(file.GetBuffer(ChunkSize));
```
To match the GDScript changes:

```suggestion
				ctx.Update(file.GetBuffer(Mathf.Min(remaining, ChunkSize)));
```

raulsntos commented on doc/classes/HashingContext.xml:
```diff
@@ -11,46 +11,52 @@
+			# Open the file to hash.
+			var file = FileAccess.open(path, FileAccess.READ)
+			# Update the context after reading each chunk.
+			while not file.get_position() &lt; file.get_len():
```
```suggestion
			while file.get_position() &lt; file.get_len():
```

raulsntos reviewed (commented):
Thanks for contributing to the documentation!

CI is complaining about indentation. Make sure the code is indented with 4 **spaces**, unlike the rest of the XML file which uses tabs.

Also, the commit is linked to a different GitHub account than the one you used to submit the PR. This is fine, but I wanted to make sure it wasn't by mistake. For more information see:  [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user)

See which "email" you have configured currently: https://github.com/godotengine/godot/pull/91920.patch

dmipeck committed:
Fix hashing context example

Example now works for any file size instead of just multiples of CHUNK_SIZE
Example also uses correct method for looping over file data

raulsntos reviewed (approved):
Sorry for the delay, I missed the notification. Looks good, thanks!

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
